src/uu/split/src/filenames.rs

```diff
@@ -0,0 +1,533 @@
+/// and increases by 2 when the chunk index becomes sufficiently
+/// large.
+///
+/// If `use_numeric_suffix` is `true`, then the `suffix` portion will
+/// be nonnegative integers. If `false`, then the `suffix` will
+/// comprise lowercase ASCII characters.
+///
+/// # Examples
+///
+/// Create filenames of the form `chunk_??.txt`:
+///
+/// ```rust,ignore
+/// use crate::filenames::FilenameFactory;
+///
+/// let prefix = "chunk_".to_string();
+/// let suffix = ".txt".to_string();
+/// let width = 2;
+/// let use_numeric_suffix = false;
+/// let factory = FilenameFactory::new(prefix, suffix, width, use_numeric_suffix);
+///
+/// assert_eq!(factory.make(0).unwrap(), "chunk_aa.txt");
+/// assert_eq!(factory.make(10).unwrap(), "chunk_ak.txt");
+/// assert_eq!(factory.make(28).unwrap(), "chunk_bc.txt");
+/// ```
+///
+/// Set `suffix_length` to 0 for filename sizes that grow dynamically:
+///
+/// ```rust,ignore
+/// use crate::filenames::FilenameFactory;
+///
+/// let prefix = String::new();
+/// let suffix = String::new();
+/// let width = 0;
+/// let use_numeric_suffix = false;
+/// let factory = FilenameFactory::new(prefix, suffix, width, use_numeric_suffix);
+///
+/// assert_eq!(factory.make(0).unwrap(), "aa");
+/// assert_eq!(factory.make(1).unwrap(), "ab");
+/// assert_eq!(factory.make(649).unwrap(), "yz");
+/// assert_eq!(factory.make(650).unwrap(), "zaaa");
+/// assert_eq!(factory.make(6551).unwrap(), "zaab");
+/// ```
+pub struct FilenameFactory {
+    additional_suffix: String,
+    prefix: String,
+    suffix_length: usize,
+    use_numeric_suffix: bool,
+}
+
+impl FilenameFactory {
+    /// Create a new instance of this struct.
+    ///
+    /// For an explanation of the parameters, see the struct documentation.
+    pub fn new(
+        prefix: String,
+        additional_suffix: String,
+        suffix_length: usize,
+        use_numeric_suffix: bool,
+    ) -> FilenameFactory {
+        FilenameFactory {
+            prefix,
+            additional_suffix,
+            suffix_length,
+            use_numeric_suffix,
+        }
+    }
+
+    /// Construct the filename for the specified element of the output collection of files.
+    ///
+    /// For an explanation of the parameters, see the struct documentation.
+    ///
+    /// # Errors
+    ///
+    /// If `suffix_length` has been set to a positive integer and `i`
+    /// is greater than or equal to the number of strings that can be
+    /// represented within that length, then this returns an error.
+    ///
+    /// For example:
+    ///
+    /// ```rust,ignore
+    /// use crate::filenames::FilenameFactory;
+    ///
+    /// let prefix = String::new();
+    /// let suffix = String::new();
+    /// let width = 1;
+    /// let use_numeric_suffix = true;
+    /// let factory = FilenameFactory::new(prefix, suffix, width, use_numeric_suffix);
+    ///
+    /// assert_true!(factory.make(10).unwrap().is_err());
+    /// ```
+    pub fn make(&self, i: usize) -> Result<String, ()> {
```

Comment:
This triggered a clippy warning on my system (Rust version 1.57):
```
warning: this returns a `Result<_, ()>`
   --> src/uu/split/src/filenames.rs:406:5
    |
406 |     pub fn make(&self, i: usize) -> Result<String, ()> {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(clippy::result_unit_err)]` on by default
    = help: use a custom `Error` type instead
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#result_unit_err
```
I think using `Option<String>` is the way to go here.

Reply:
Or you could return `UResult` with the `USimpleError` here. Whatever you prefer.

Reply:
I'll make that change, thanks.